### DISCUSSION TITLE
Cannot Create Output

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [statiqdev/Discussions#80] [Q&A] [answered]
Hey @daveaglick, seems your last patch didn't work. I misplaced the issue else I would have responded there.

You can see from this [action result](https://github.com/LuzFaltex/luzfaltex.com/runs/2865608652?check_suite_focus=true) that we're still running into the same issue

```
Content/PostProcess » RenderContentPostProcessTemplates » ExecuteIf » MirrorResources » Cannot create 'D:\a\luzfaltex.com\luzfaltex.com\output\mirror\www.luzfaltex.com' because a file or directory with the same name already exists.
```

@Inerska sent me an email about this - apparently they're seeing the same issue.

I believe I'm running the latest version of everything:
```xml
<ItemGroup>
  <PackageReference Include="Statiq.Razor" Version="1.0.0-beta.46" />
  <PackageReference Include="Statiq.Sass" Version="1.0.0-beta.46" />
  <PackageReference Include="Statiq.Web" Version="1.0.0-beta.30" />
</ItemGroup>
```

Any ideas as to what may be causing this? Here's the [build script](https://github.com/LuzFaltex/luzfaltex.com/blob/dev/.github/workflows/Development.yml) if you're curious what I'm running.

Edit: As last time, I performed the same operation locally using `dotnet run -- deploy` after completely deleting the output folder. The same error message appears. In the mirror folder, I still get the `www.luzfaltex.com` file (rather than a directory).

If I run with the `preview` flag instead, then the mirror folder ends up with the directories I would expect. I have preserved the `www.luzfaltex.com` file for reference.

![output folder](https://cdn.discordapp.com/attachments/468748717684686868/855854950680756234/unknown.png)

**Comment by TheBrambleShark** [comment, marked as answer]
I believe I found the answer. Earlier in my log, visible in debug mode, I have this line:

> Downloading resource from https://www.luzfaltex.com/assets/img/favicon.svg to mirror/www.luzfaltex.com/assets/img/favicon.svg

For some reason, it thinks this is an external site. The asset is being defined as so:

```cshtml
@{
    var favicon = ExecutionContext.GetLink("assets/img/favicon.svg", true);
}
<!doctype html>
<html lang="en">
<head>
    <!-- snip! -->
    <link rel="icon" href="@favicon">
    <!-- snip! -->
</head>
```

Edit: setting the `includeHost` parameter of `Execution.GetLink()` to `false` resolves the issue. Looks like this may be where the issue is coming into play. For some reason, even though the `Host` property is set to `www.luzfaltex.com`, it treats that domain as being external and tries to download content from it anyways. I had somewhat bypassed it by uploading my favicon to the site that currently exists on that domain, but didn't stop to think why it wanted me to do that.

After setting all values there to false, the deploy pipeline now finishes successfully.

**Reply by daveaglick** [reply]
The good news is that I think what you described is exactly what was patched. 

The Framework versions you have in the package references are the latest, but Statiq.Web is a version behind - the latest there is 1.0.0-beta.31, so the older Statiq.Web reference is likely bringing in the older Statiq.Core as well. FWIW, Statiq.Razor and Statiq.Sass are also referenced by Statiq.Web so that's hopefully the only reference you need. Go ahead and update Statiq.Web to 1.0.0-beta.31 and let me know if that resolves the problem!

**Reply by TheBrambleShark** [reply]
Sorry for getting back so much later-

That seemed to work. Now I'm just fighting with an unrelated "not found" issue from the GitHub piece.

**Reply by daveaglick** [reply]
Great, glad it worked!
